packages/plugin-core/src/components/doctor/utils.ts

```diff
@@ -88,7 +90,30 @@ export class DoctorUtils {
           noteA: duplicate,
           noteB: note,
         });
-        VSCodeUtils.showMessage(MessageSeverity.WARN, error.message, {});
+        VSCodeUtils.showMessage(
+          MessageSeverity.WARN,
+          error.message,
+          {},
+          { title: "Fix It" }
+        ).then((resp) => {
+          if (resp && resp.title === "Fix It") {
+            const cmd: vscode.Command = {
```

Comment:
the doctor action probably needs to force a reload after it's run (or make sure the result of the action actually emits the correct note change entry once it's done).

Depending on the note's filename, there will be cases where note A is recognized by the engine or note B, but not at the same time (since we don't allow duplicate ids in the engine). Where you run the doctor command would make a difference I think.

If you ran it in a note that wasn't in the engine state, it would ideally be a `create`, but if you ran it in a note that was in the engine, it would be a `delete` followed by a `create`.

These might be some possible reasons why the note changes are not detected in the test you added.

Reply:
Did some debugging, it looks like the test is completed before the doctor actions gets completed. Thanks for the help!